packages/ai-jsx/src/batteries/docs.tsx

```diff
@@ -1,154 +1,498 @@
+import { Embeddings } from 'langchain/embeddings/base';
+import { OpenAIEmbeddings } from 'langchain/embeddings/openai';
+import { TokenTextSplitter } from 'langchain/text_splitter';
+import _ from 'lodash';
+import { similarity } from 'ml-distance';
+import pino from 'pino';
+import { Jsonifiable } from 'type-fest';
+import { ChatCompletion, SystemMessage, UserMessage } from '../core/completion.js';
 import * as LLMx from '../index.js';
 import { Node } from '../index.js';
-import { ChatCompletion, SystemMessage, UserMessage } from '../core/completion.js';
-import { Jsonifiable } from 'type-fest';
-import { TokenTextSplitter } from 'langchain/text_splitter';
-import { MemoryVectorStore } from 'langchain/vectorstores/memory';
-import { Document as LangChainDocument } from 'langchain/document';
-import { Embeddings } from 'langchain/embeddings/base';
+
+// DO_NOT_SUBMIT
+const pinoLogger = _.once(() =>
+  // @ts-expect-error
+  pino(
+    { name: 'ai-jsx-docs', level: 'trace' },
+    // N.B. pino.destination is not available in the browser
+    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+    pino.destination?.({
+      dest: './ai-jsx-docs.log',
+      sync: true, // Synchronous logging
+    })
+  )
+)();
```

Comment:
@petersalas, I have [these docs](https://ai-jsx-docs-m4s36woh3-fixie-ai.vercel.app/guides/observability#logging-from-within-components) for how to log within a component, but what are we thinking for this context, where we're not immediately within a component? Maybe this is where we test our "can everything be a component" question? It seems a little clumsy for components to have to pass the logger into every helper they call.

Reply:
FWIW, I don't intend to merge the logging into main, at least as currently written. I just haven't removed it yet because it's useful for testing and I expect there to be other comments. :)

Reply:
Certainly crawling doesn't make sense to be wrapped as component, in that the machinery that drives crawling is ~orthogonal to rendering/query-time machinery -- so I don't even think there's a sensible `Logger` to pass. (Though in general if we want logging to be associated with a render, I think passing the `Logger` explicitly is sensible?)

In general it's not obvious to me how exactly crawling code can/should fit into AI.JSX (vs. say a Fixie SDK).

Reply:
I get the angle about "this isn't related to a render", but I think we want the ability to write to the common log stream, even outside of things related to a render, like what we see here.

The point about "maybe this belongs outside of AI.JSX and in a Fixie SDK" is a good one – maybe eventually we'll split this type of logic out into a separate module?